### PR TITLE
Fixed memory leak and some -Werror=unused-result warnings

### PR DESCRIPTION
**Pull request by OMGtechy**
Signed-off-by: Joshua Gerrard <enquiries@joshuagerrard.com>

### CODE REVIEW

File: AUTHORS [hunk 1]
```diff
@@ -165,6 +165,8 @@ Josh Hursey, Indiana University, Oak Ridge National Laboratory, Los Alamos Natio
   jjhursey@open-mpi.org
 Joshua Gerrard
   enquiries@joshuagerrard.com
+  technopiano@gmail.com
+  joshuagerrard+ompi-commit@protonmail.com
```
Comment (jsquyres) [inline AUTHORS]: @OMGtechy Are these the emails that you committed with?  We typically only show the emails that show up in `git log`.

File: ompi/errhandler/errhandler_predefined.c [hunk 2]
```diff
@@ -177,21 +177,30 @@ static void backend_fatal_aggregate(char *type,
...
     arg = va_arg(arglist, char*);
     va_end(arglist);
 
-    asprintf(&prefix, "[%s:%d]", ompi_process_info.nodename,
-             (int) ompi_process_info.pid);
+    if (asprintf(&prefix, "[%s:%d]",
+                 ompi_process_info.nodename,
+                 (int) ompi_process_info.pid) == -1) {
+        // non-fatal, we could still go on to give useful information here...
+        opal_output(0, "%s", "Could not write node and PID to prefix");
```
Comment (jsquyres) [inline ompi/errhandler/errhandler_predefined.c]: Just curious -- why do the double-string form?  You shouldn't need to use the copy through %s -- you could just put the 3rd argument directly as the 2nd arg.

File: ompi/errhandler/errhandler_predefined.c [hunk 3]
```diff
@@ -177,21 +177,30 @@ static void backend_fatal_aggregate(char *type,
...
 
     if (NULL != error_code) {
         err_msg = ompi_mpi_errnum_get_string(*error_code);
         if (NULL == err_msg) {
-            err_msg_need_free = true;
-            asprintf(&err_msg, "Error code: %d (no associated error message)",
-                     *error_code);
+            if (asprintf(&err_msg, unknown_error_code,
+                         *error_code) == -1) {
+                opal_output(0, "%s", "Could not write to err_msg");
```
Comment (jsquyres) [inline ompi/errhandler/errhandler_predefined.c]: Same comment as above -- %s doesn't seem to be needed...?

File: ompi/errhandler/errhandler_predefined.c [hunk 4]
```diff
@@ -177,21 +177,30 @@ static void backend_fatal_aggregate(char *type,
...
+    }
 
     if (NULL != error_code) {
         err_msg = ompi_mpi_errnum_get_string(*error_code);
         if (NULL == err_msg) {
-            err_msg_need_free = true;
-            asprintf(&err_msg, "Error code: %d (no associated error message)",
-                     *error_code);
+            if (asprintf(&err_msg, unknown_error_code,
+                         *error_code) == -1) {
```
Comment (jsquyres) [inline ompi/errhandler/errhandler_predefined.c]: Will a failed `asprintf()` leave NULL in `err_msg`?  Or could it be some random value (that `free(err_msg)`, below, will choke on)?

File: ompi/errhandler/errhandler_predefined.c (hunk 2 above)
Comment (OMGtechy) [inline ompi/errhandler/errhandler_predefined.c]: It's a habit to stop situations like this:

`opal_output(0, user_input, ...)`

Quite happy to change it if you want though :)

File: AUTHORS (hunk 1 above)
Comment (OMGtechy) [inline AUTHORS]: One of them is - I've updated accordingly.

File: ompi/errhandler/errhandler_predefined.c (hunk 4 above)
Comment (OMGtechy) [inline ompi/errhandler/errhandler_predefined.c]: Good spot! I also made the same fix for `prefix`

**Comment by OMGtechy** [comment]
> Sorry for the delay in reviewing -- December gets busy!

No worries! Your review comments should be addressed now.

File: ompi/errhandler/errhandler_predefined.c (hunk 2 above)
Comment (jsquyres) [inline ompi/errhandler/errhandler_predefined.c]: FWIW, I think compilers only warn on that for libc functions (e.g., printf).  They don't for opal_output.  But it doesn't matter that much...

**Pull request merged**